Engine/media/audio/audio.cpp

```diff
@@ -634,23 +699,26 @@ void shutdown_sound()
 
 // the sound will only be played if there is a free channel or
 // it has a priority >= an existing sound to override
-int play_sound_priority (int val1, int priority) {
+static int play_sound_priority (int val1, int priority) {
     int lowest_pri = 9999, lowest_pri_id = -1;
 
+    AudioChannelsLock lock;
+
     // find a free channel to play it on
     for (int i = SCHAN_NORMAL; i < MAX_SOUND_CHANNELS; i++) {
```

Comment:
Potential for AudioChannelsLock within loop? We interact with another channel but it's later.